src/identify/index.js

```diff
@@ -118,8 +118,8 @@ function updateSelf (peerSelf, observedAddr) {
   }
 
   for (var i = 0; i < peerSelf.previousObservedAddrs.length; i++) {
-    if (peerSelf.previousObservedAddrs[i].toString() === omh.toString()) {
-      peerSelf.previousObserveredAddrs.splice(i, 1)
+    if (multiaddr(peerSelf.previousObservedAddrs[i]).toString() === omh.toString()) {
```

Comment:
I didn't change what is stored in `previousObservedAddrs`. The issue was that when inspecting `observedAddr` it is a binary buffer and not a string buffer, so this equality check would always fail. So maybe https://github.com/Dignifiedquire/node-ipfs-swarm/blob/identify-tests/src/identify/index.js#L128 should be changed to 
```js
```